mcs/class/System.Private.CoreLib/System.Collections.Generic/EqualityComparer.cs

```diff
@@ -47,6 +47,16 @@ static EqualityComparer<T> CreateComparer ()
 			
 			return new ObjectEqualityComparer<T>();
 		}
+
+		internal virtual int IndexOf (T[] array, T value, int startIndex, int count)
```

Comment:
Why virtual?

Reply:
So it can have optimized implementations later.